cpp/src/gandiva/gandiva_object_cache.h

```diff
@@ -0,0 +1,61 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#ifndef ARROW_GANDIVA_OBJECT_CACHE_H
+#define ARROW_GANDIVA_OBJECT_CACHE_H
+
+#include <llvm/Support/MemoryBuffer.h>
+#include "llvm/ExecutionEngine/ObjectCache.h"
+#include "llvm/IR/Module.h"
+#include "gandiva/cache.h"
+
+namespace gandiva {
+template<class CacheKey>
+class GandivaObjectCache : public llvm::ObjectCache {
```

Comment:
It would be good to add a comment explaining the purpose of that class

Reply:
Done